internal/webhook/bootstrapprovider_webhook.go

```diff
@@ -47,18 +48,18 @@ var (
 )
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
-func (r *BootstrapProviderWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) error {
-	return nil
+func (r *BootstrapProviderWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	return admission.Warnings{}, nil
```

Comment:
I think we [can](https://github.com/kubernetes-sigs/cluster-api/pull/8007/files#diff-8cd1f8530fe1126f54a0febcd002685b7dc898fc3b983941450f5974aa503be9R76) just return `nil` instead of `admission.Warnings{} ` here and other places?